### PR TITLE
Redo take 2

### PR DESCRIPTION
**Pull request by alan-andrade**
Simple skeleton which will do exception handling, callback to return results, 2 pruebas.

🚴

### CODE REVIEW

File: test/autobotResource.js [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+  constructor(data) {
+    super(data);
+  }
+}
+
+describe('ResourceBot', function() {
+  var data = { hello: 'world' };
+  var getInfo = function(input, data, handler) {
+    data_field = input.args[0]
+    handler.success(data[data_field]);
```
Comment (alan-andrade) [inline test/autobotResource.js]: This `data` is being grabbed from the outer scope, not the function param.

File: main/autobot/project_tracker.js [hunk 2]
```diff
@@ -0,0 +1,37 @@
+var Handler = require('../autobot/handler');
+var JiraApi = require('jira').JiraApi;
+
+var JiraTracker = function (config) {
```
Comment (abMatGit) [inline main/autobot/project_tracker.js]: I feel like this should live on its own, along with each other potential tracker. This way we aren't importing a crap ton of configurations for each tracker within one file. (configurations that live from line 15-> 18.

File: configs/jira_credentials.js [hunk 3]
```diff
@@ -0,0 +1,5 @@
+module.exports = {
+    username: 'your_username',
+    password: 'password',
+    host: 'host'
+}
```
Comment (abMatGit) [inline configs/jira_credentials.js:7]: 👍 👍 👍 👍

File: main/autobot/project_tracker.js [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+        host: 'lumoslabs.atlassian.net',
+        port: null,
+        user: 'alan@lumoslabs.com',
+        password: ''
+    });
+}
+
+ProjectTracker.prototype = {
+    getStory: function (storyId, callback) {
+        var handler = new Handler(callback);
```
Comment (abMatGit) [inline main/autobot/project_tracker.js:9]: Hmmmmm we should maybe think about a place for these callbacks and maybe how to group them. 
[The status callback](https://github.com/abMatGit/hackathon-2016/pull/6/files#diff-e83dd419bac6347c97cc0f70a0bd1e22R26) is one example of many that we'll end up using. For the moment though i think they can all just live in this file or something.

**Comment by alan-andrade** [comment]
# Design
- `ProjectTracker` is the base class for future Trackers. Jira is currently the only Tracker.
- `ProjectTracker` interacts with the real library through the `client` property.
- We still need to define the interface for that client. So far it has the `node_jira` interface.
- merge if you like, or modify at will.

File: main/autobot/project_tracker.js (hunk 4 above)
Comment (alan-andrade) [inline main/autobot/project_tracker.js:9]: I didn't get the idea of the status callback. I imagine we can always pass around the autobot standard `Handler`.

**Pull request merged**